src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -32,17 +36,26 @@ public override TextDocumentAttributes GetTextDocumentAttributes(DocumentUri uri
 
         public override Task<Unit> Handle(DidChangeTextDocumentParams request, CancellationToken token)
         {
+            DocumentUri documentUri = request.TextDocument.Uri;
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            if (string.Equals(Path.GetFileName(documentUri.Path), LanguageConstants.BicepConfigSettingsFileName))
+            {
+                // Refresh compilation of source files in workspace when local bicepconfig.json file is edited
+                BicepConfigChangeHandler.RefreshCompilationOfSourceFilesInWorkspace(compilationManager, documentUri.ToUri(), workspace, contents);
+            }
+            else
+            {
+                this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+            }
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams request, CancellationToken cancellationToken)
         {
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId, reloadBicepConfig: true);
```

Comment:
>reloadBicepConfig [](http://example.com/codeflow?start=170&length=17)

Do we really want to blindly reload the config every time a file is open?

Reply:
Reverted this change here:
https://github.com/Azure/bicep/commit/1ee10711ad7bdd7412967c1ce6ae6f6a1923fa5b